### PR TITLE
Only Support Custom Pokemon icons in config

### PR DESCRIPTION
**Pull request by versx**
Removes the need to create new icon repositories. Only allows for custom Pokemon icons via the config for remote and local repos.

Remote repositories are expected to look like the following with an `index.json` to depict the available form icons:  
https://github.com/Mygod/pokicons/tree/master/v1
`index.json` example:  
https://raw.githubusercontent.com/Mygod/pokicons/master/v1/index.json

Config example:
image

### CODE REVIEW

File: src/config.example.json [hunk 1]
```diff
@@ -159,21 +159,8 @@
...
-        "RemotePokemonExample": {
-            "path": "https://example.com/pokemon_images",
-            "pokemonList": ["001_163", "002_166", "002_166_shiny", "003_169", "003_169_female"],
-            "raidOffsetY": 0.269,
-            "questOffsetY": 0,
-            "gymAnchorY": 0.849,
-            "pokestopAnchorY": 0.896
-        }
+        "Default": "/img/pokemon",
+        "POGO": "https://raw.githubusercontent.com/Mygod/pokicons/master/v1"
```
Comment (versx) [inline src/config.example.json:163]:
```suggestion
        "Default": {
            "path": "/img"
        },
        "POGO": {
            "path": "https://raw.githubusercontent.com/Mygod/pokicons/master/v1"
        }
```

File: src/data/map.js [hunk 2]
```diff
@@ -1058,7 +1058,7 @@ const getNests = async (minLat, maxLat, minLon, maxLon, nestFilterExclude = null
     return null;
 };
 
-const getSearchData = async (lat, lon, id, value) => {
+const getSearchData = async (lat, lon, id, value, iconStyle) => {
```
Comment (Mygod) [inline src/data/map.js:1061]: Stop letting server resolve iconStyle. 🤷

Comment (Mygod) [reply src/data/map.js:1061]: There is no fallback logic on the server side. This will lead to form not resolving correctly.

Comment (versx) [reply src/data/map.js:1061]: How else will the server know what style the client chose?

File: src/config.example.json [hunk 3]
```diff
@@ -160,19 +160,10 @@
     },
     "icons": {
         "Default": {
-            "path": "/img",
-            "raidOffsetY": 0.269,
-            "questOffsetY": 0,
-            "gymAnchorY": 0.849,
-            "pokestopAnchorY": 0.896
+            "path": "/img"
```
Comment (Mygod) [inline src/config.example.json:163]: Wrong path?

File: src/data/map.js (hunk 2 above)
Comment (Mygod) [reply src/data/map.js:1061]: No need. Send pokemon data to client and let client do the work. See my other PR on quest rewards.

File: src/config.example.json (hunk 3 above)
Comment (versx) [reply src/config.example.json]: Thanks

**Review by clburlison — APPROVED** [review]
Works for me. If we're going to force this style I would prefer a link to the [processor](https://github.com/Mygod/pokemon-icon-postprocessor) so image repo maintainers can at least have a chance at maintaining this.

File: src/data/map.js (hunk 2 above)
Comment (Mygod) [reply src/data/map.js:1061]: For example: https://github.com/versx/MapJS/pull/117/files#diff-095e8a9344f3ac4f7d1799f877e63e05L225

**Pull request merged**